### DISCUSSION TITLE
Reduce amount of opened connections

### DISCUSSION THREAD

**Discussion by Timelessprod** [apache/airflow#40276] [Q&A]
Hello,

I use Airflow with Postgres databases on GCP and I noticed that it opens a lot of new connections instead of relying on a same pool. We often reach more than 1,200 connections opened every 2 minutes (both on jobs' DBs and Airflow internal DB) which is a lot and our proxy sometimes drop those connections, leading to tasks failure.

Is there any way to reduce the amount of simultaneous connections or to set Airflow to use a connection pool ?

Thanks

**Comment by potiuk** [comment]
Use pgbouncer. This is recommended production setting for Postgres - see documentation about database setup (search for pgbouncer in the docs), it is explained there.
